docs/serving/using-knative-services.md

```diff
@@ -0,0 +1,232 @@
+kind: Service
+metadata:
+ name: helloworld-go # (2)
+ namespace: default # (3)
+spec:
+ template:
+  spec:
+   containers:
+    - image: gcr.io/knative-samples/helloworld-go # (4)
+     env:
+      - name: TARGET # (5)
+       value: "Go Sample v1"
+```
+**(1)** Current version of Knative   
+**(2)** The name of the app   
+**(3)** The namespace the app will use   
+**(4)** The URL to the image of the app   
+**(5)** The environment variable printed out by the sample app
+
+> Note: If you’re deploying an image of your own app, update the name of the app and the URL of the image accordingly.
+
+## Creating a private cluster-local Service
+By default services deployed through Knative are published to an external IP
+address, making them public services on a public IP address and with a
+[public URL](./using-a-custom-domain.md).
+
+While this is useful for services that need to be accessible from outside of the
+cluster, frequently you may be building a backend service which should not be
+available off-cluster.
+
+Knative provides two ways to enable private services which are only available
+inside the cluster:
+
+1. To make all services only available only from within the cluster, change the default domain to
+   `svc.cluster.local` by
+   [editing the `config-domain` config map](./using-a-custom-domain.md). This
+   will change all services deployed through Knative to only be published to the
+   cluster, none will be available off-cluster.
+1. To make an individual service cluster-local, the service or route can be
+   labeled in such a way to prevent it from getting published to the external
+   gateway.
+
+### Label a service to be cluster-local
+
+To configure a Service to only be available on the cluster-local network (and
+not on the public Internet), you can apply the
+`serving.knative.dev/visibility=cluster-local` label to the Service object.
+
+To label the Service:
+
+```shell
+kubectl label kservice ${KSVC_NAME} serving.knative.dev/visibility=cluster-local
+```
+
+To label a route:
+
+```shell
+kubectl label route ${ROUTE_NAME} serving.knative.dev/visibility=cluster-local
+```
+
+For example, you can deploy the [Hello World sample](./samples/helloworld-go)
+and then convert it to be an cluster-local service by labeling the service:
+
+```shell
+kubectl label kservice helloworld-go serving.knative.dev/visibility=cluster-local
+```
+
+You can then verify that the change has been made by verifying the URL for the
+helloworld-go service:
+
+```shell
+kubectl get ksvc helloworld-go
+
+NAME            URL                                              LATESTCREATED         LATESTREADY           READY   REASON
+helloworld-go   http://helloworld-go.default.svc.cluster.local   helloworld-go-2bz5l   helloworld-go-2bz5l   True
+```
+
+The service returns the a URL with the `svc.cluster.local` domain, indicating
+the service is only available in the cluster local network.
+
+## Deploying an application using Knative Services
```

Comment:
From here down i think ALL but "Modifying Knative Services", is out of scope for this "Creating a Knative service" task and should move back into the "Getting Started" guide.